cmd/object-handlers.go

```diff
@@ -2882,6 +2882,18 @@ func (api objectAPIHandlers) DeleteObjectHandler(w http.ResponseWriter, r *http.
 			return
 		}
 		if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
+			// Send an event when the object is not found
+			objInfo.Name = object
+			objInfo.VersionID = opts.VersionID
+			sendEvent(eventArgs{
+				EventName:    event.ObjectRemovedDelete,
```

Comment:
AWS S3 creates a DEL marker when DELETE is called on a non-existing object. Thinking if we can send `s3:ObjectRemoved:DeleteMarkerCreated` for such requests on versioned buckets.. 

any suggestions? @harshavardhana

Reply:
MinIO does not create a delete marker in this case, so we cannot send a notification with reliable information, such as delete-marker version-id, does someone need this ?

Reply:
We should send notification that we didn't do anything that's all. 

404 or otherwise we should notify the 204s sent to the client.

In this case nothing is created so it is a no-op of sort where we say that it's 204 but we didn't have the object in the first place.

Add an additional metadata to the notification.